### ISSUE TITLE
Batched Linear Algebra Support

### ISSUE THREAD

**Issue by sslattery** [ORNL-CEES/DataTransferKit#283]
Batched linear algebra is needed for some point cloud algorithms. Identify the path forward for some support batched LAPACK operations. Does not seem to be in [kokkos-kernels](https://github.com/kokkos/kokkos-kernels) yet.

Supported features needed are in the 2.0 implementation of the [moving least square operator](https://github.com/ORNL-CEES/DataTransferKit/blob/dtk-2.0/packages/Operators/src/PointCloud/DTK_LocalMLSProblem_impl.hpp). These include the LAPACK functions:

- [ ] GETRF (LU Factorization)

- [ ] GECON (condition number estimate)

- [ ] GELSS (least square solution to system of equations)

- [ ] GESVD (singular value decomposition)

Resolution of this issue should identify/create a capability in which these functions are available as KOKKOS_INLINE_FUNCTIONS

**Comment by sslattery** [comment]
Should expect the following in the August kokkos-kernels release via update from kokkos-kernels team: 

- GETRF (LU Factorization)

- GEMM (Matrix-matrix multiply)

- TRSM (Lower/Upper triangular solve)

Which means that at the moment we will be limited to moving least square basis systems that can be factored using LU

**Comment by aprokop** [comment]
I don't think this is going to happen in FY17. Can we reassign it to a different milestone?
